solc/CommandLineInterface.cpp

```diff
@@ -499,13 +515,38 @@ bool CommandLineInterface::processInput()
 
 	function<pair<string,string>(string const&)> fileReader = [this](string const& _path)
 	{
-		auto path = boost::filesystem::path(_path);
-		if (!boost::filesystem::exists(path))
+		// Try to find the longest prefix match in all remappings. At the end, there will be an
+		// empty remapping so that we also try the path itself.
+		int errorLevel = 0;
+		size_t longestPrefix = 0;
+		string bestMatchPath;
+		for (auto const& redir: m_remappings)
+		{
+			auto const& virt = redir.first;
+			if (longestPrefix > 0 && virt.length() <= longestPrefix)
+				continue;
+			if (virt.length() > _path.length() || !std::equal(virt.begin(), virt.end(), _path.begin()))
+				continue;
+			string path = redir.second;
+			path.append(_path.begin() + virt.length(), _path.end());
```

Comment:
I think the problem only happens if you use a different style on the left of the `=` as you do on the right. So `solc virtual/path=real/path x.sol` works, as does `solc virtual/path/=real/path/ x.sol`, but `solc virtual/path/=real/path` x.sol` might not. I think we should not assume that the remappings are actually paths. It could also be something like `solc ipfs:=/var/ipfs-fuse/ x.sol`.